uniconnect-epfl/uniconnect
# feat: setup firestore database with an initial interaction

# What I did

- Setup the Firebase Firestore Database
-  Implemented a write function to a database collection when registering for the first time. 

Now, when users register, their email is stored in Firestore and their email + password on Firebase Auth.

# How I did it

- I added this dependency `firebase/firestore`
- Instantiated a firestore Database in `firebaseConfig.ts`
- wrote some logic in `Registration.ts`
- linked everything together in the Authentication Screen

# How to verify it
- 1. start Expo Go
- 2. Go through the signup phase and onto the Auth Screen
- 3. Register with a new email
- 4. Press on the `Send confirmation e-mail` button. 
- 5. Go to Firebase conssole and look into the Firestore collection and Firebase Authentication, you'll see the new email


# Demo video
There is no demo video as it is a pure backend functionality.

# Testing
I have changed tests so that we separate frontend and backend unit testing. Thus, the two backend functions of the Authentication Screen (`createAccount` and `storeEmail`) have been mocked. The tests have been modified to test the functionality of the overall `submitForm` function.

Thus, an upcoming PR should be to test the backend functionalities thoroughly.

# Pre-merge checklist
The changes I have introduced:
